File: view/view-farma/src/app/interface/entry-product.ts
import { Entry } from "./entry"; // Importe a interface de Entry

export interface EntryProduct {
  id?: number; // Em TypeScript, 'number' corresponde a Long e outras formas de números inteiros
  entry?: Entry; // Relacionamento ManyToOne com Entry
  codLot?: string;
  description?: string;
  unit?: string;
  amount?: number; // 'int' em Java corresponde a 'number' em TypeScript
}

File: src/main/java/com/horus/formataArquivoHorus/model/entity/EntryProduct.java
package com.horus.formataArquivoHorus.model.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "entry_product_tb")
@Data
public class EntryProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "entry_id", referencedColumnName = "id")
    private Entry entry;

    @Column(name = "codLot")
    private String codLot;

    @Column(name = "description")
    private String description;

    @Column(name = "unit")
    private String unit;

    @Column(name = "amount")
    private int amount;
}

